accounts-db/src/account_storage.rs

```diff
@@ -368,10 +368,10 @@ impl<'a> AccountStoragesOrderer<'a> {
 }
 
 impl Index<usize> for AccountStoragesOrderer<'_> {
-    type Output = AccountStorageEntry;
+    type Output = Arc<AccountStorageEntry>;
```

Comment:
I think this can be avoided, in generate index code you don't *really* need the arc, you can store index in the vec and then look it up directly from the original array, see my attempt to introduce scoped thread https://github.com/kskalski/agave/pull/4/files

Reply:
Also, I'm working on having scan in threads as utility function here:
https://github.com/anza-xyz/agave/pull/7828

the API will need to be extended:
* with the indices as in https://github.com/anza-xyz/agave/pull/7821
* with prefetch functionality as in https://github.com/anza-xyz/agave/pull/6878